### ISSUE TITLE
Custom Icons on Folder/Connection

### ISSUE THREAD

**Issue by FR-TheS** [mRemoteNG/mRemoteNG#642]
|Operating system | Windows 7 x64 |
|mRemoteNG version| 1.75.7008.17935  |

Can you add the feature to add custom icons and the possibility to put icon on Folders too ?
It will be great to see the difference between environment(production, bench, etc...)

Example on an another tool :
![image](https://user-images.githubusercontent.com/21204359/28273528-ee648d4e-6b0e-11e7-9693-49dcf5c6b6de.png)

**Comment by Fiiti** [comment]
+1 A very missing feature for me also.

**Comment by fred9176** [comment]
+1

**Comment by kpolak** [comment]
+1

**Comment by gdsotirov** [comment]
+1 Would be especially useful with large connection lists like mine, which is constantly growing. :-)

**Comment by vmario89** [comment]
yep +1

**Comment by haat-reassured** [comment]
+1

**Comment by viatek-andy** [comment]
+1

**Comment by viatek-andy** [comment]
I don't know since when, but connection icons can be customized. I can't see a way to do folder icons, though. 
Just drop the custom icons into the `..\mRemoteNG\icons` folder. That makes them available in the custom icons list. (running 1.76.8.17214).

**Comment by dboc** [comment]
Cool! But i need to change folder icons.

**Comment by gdsotirov** [comment]
Yes, dropping icons to `mRemoteNG\icons` works and I use it, but it would be nice to be able to customize folder icons as well especially if one has many connections.

**Comment by Zewwy** [comment]
Whens this becoming a thing? I was just about to open an Issue ticket and saw this one open. :D

**Comment by Greycat01** [comment]
Does adding custom icons create an issue when those icons aren't present in a new location? Say after export/import

**Comment by peterj04** [comment]
What I've noticed is that the connection remains intact, but the icon is gone until I add the custom icons back to the icons folder, so I always make a backup of my custom icons just in case I need to reinstall.

> Does adding custom icons create an issue when those icons aren't present in a new location? Say after export/import

**Comment by Anders-Ryden** [comment]
+1
Folder icons would be really useful

**Comment by noz482** [comment]
The ability to use custom icons for folders would help me out a ton. +1

Thanks for the great application. Makes my role as a lab admin much easier.

**Comment by Logico68** [comment]
absolutely +1

**Comment by itb-dev-bw** [comment]
I also would like to have an easier way of setting custom connection icons! +1

Also, I'm missing a way to use custom icons that **doesn't need administrator access** to write into Program Files. Better to use %LocalAppData% or save icons directly into the connection file (like KeePass does it).

**Comment by robertpopa22** [comment]
**Fix available for testing**

**Commit:** [`7e27a438`](https://github.com/robertpopa22/mRemoteNG/commit/7e27a4389ccee4e5c024e937c723f33111c2bdd8)
**Branch:** `main`
**What changed:** Add a CustomIconPath property to ConnectionInfo, allow embed

**Download latest beta:** [v1.81.0-beta.2](https://github.com/robertpopa22/mRemoteNG/releases/tag/v1.81.0-beta.2)

Please test and report if this resolves your issue.

---
_Automated by mRemoteNG Issue Intelligence System_

**Comment by robertpopa22** [comment]
**Correction:** The download link in my previous comment was broken. Here is the correct one:

**Download pre-built binaries:** https://github.com/robertpopa22/mRemoteNG/releases

Look for the latest **v1.81.0-beta.2** build (x64, x86, ARM64 — both portable and self-contained variants available).

Sorry for the inconvenience!
